modules/core/04-channel/keeper/upgrade.go

```diff
@@ -81,15 +81,30 @@ func (k Keeper) ChanUpgradeTry(
 	return types.Upgrade{}, nil
 }
 
-// WriteUpgradeTryChannel writes a channel which has successfully passed the UpgradeTry step.
+// WriteUpgradeTryChannel writes a channel which has successfully passed the UpgradeTry handshake step.
 // An event is emitted for the handshake step.
 func (k Keeper) WriteUpgradeTryChannel(
 	ctx sdk.Context,
 	portID, channelID string,
 	proposedUpgrade types.Upgrade,
+	flushStatus types.FlushStatus,
 ) {
-	// TODO
-	// grab channel inside this function to get most current channel status
+	defer telemetry.IncrCounter(1, "ibc", "channel", "upgrade-try")
+
+	currentChannel, found := k.GetChannel(ctx, portID, channelID)
+	if !found {
+		panic(fmt.Sprintf("could not find existing channel when updating channel state in successful ChanUpgradeTry step, channelID: %s, portID: %s", channelID, portID))
+	}
+
+	previousState := currentChannel.State
```

Comment:
if we will be going ahead with https://github.com/cosmos/ibc/pull/978 then we can remove this var and hardcode to `INITUPGRADE` in the log statement below.

we can handle this later